### PR TITLE
ref #231 Add keyspace notifications to all commands

### PR DESCRIPTION
**Pull request by gkorland**

### CODE REVIEW

File: src/module.c [hunk 1]
```diff
@@ -845,6 +871,9 @@ int TSDB_get(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
 
     ReplyWithSeriesLastDatapoint(ctx, series);
     RedisModule_CloseKey(key);
+
```
Comment (danni-m) [inline src/module.c:874]: @gkorland why notify on get?

File: src/module.c [hunk 2]
```diff
@@ -826,6 +849,9 @@ int TSDB_incrby(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     int rv = internalAdd(ctx, series, currentUpdatedTime, result, DP_LAST);
     RedisModule_ReplicateVerbatim(ctx);
     RedisModule_CloseKey(key);
+
+    RedisModule_NotifyKeyspaceEvent(ctx, REDISMODULE_NOTIFY_GENERIC, "ts.incrby", key);
```
Comment (danni-m) [inline src/module.c:853]: This can also be decryby

File: src/module.c (hunk 1 above)
Comment (gkorland) [reply src/module.c:874]: right... over motivation.... I'll remove it

File: src/module.c (hunk 2 above)
Comment (gkorland) [reply src/module.c]: :+1:

**Comment by filipecosta90** [comment]
@gkorland with regards to the keyspace notifications we should definitely patch test this prior merging given it might deeply affect performance when notify-keyspace-events is enabled ( the TS.MADD  example is a great one, in which one command from client can generate a ton of notifications.... ).
Adding the `requires-patch-benchmark` label.

**Review by danni-m — APPROVED** [review]

**Comment by filipecosta90** [comment]
@gkorland @danni-m  #732 will address the question of what's the impact of disabled/enabled keyspace notifications with no subscribers.

**Pull request merged**